usr/lib/icsf_stdll/icsf.c

```diff
@@ -906,8 +932,30 @@ static int icsf_ber_put_attribute_list(BerElement * ber, CK_ATTRIBUTE * attrs,
                                        CK_ULONG attrs_len)
 {
     size_t i;
+    CK_OBJECT_CLASS class = CK_UNAVAILABLE_INFORMATION;
+
+    find_ulong_attribute(attrs, attrs_len, CKA_CLASS, &class);
```

Comment:
it's a bit confusing that routine find_ulong_attribute() has a rc, but the rc is not checked. However, looks like the class value is set in all cases, in some just the preset value is still set after calling the routine.

Reply:
Right, if the function fails, then the preset value is still set and the switch below goes into the default case.

Reply:
But then, why not make the function return void?

Reply:
`find_ulong_attribute()` is pretty much a copy from `find_bbool_attribute()` in usr/lib/common/utility.c. I first wanted to put `find_ulong_attribute()` also into usr/lib/commno/utility.c, but module icsf.c where usr/lib/commno/utility.c. is used is also lined into the usr/sbin/pkcsicsf/pkcsicsf tool, and there utility.c is not included. Trying to include it lead to further problems (other missing functions, etc), so I decided to make `find_ulong_attribute`  a static function in icsf.c directly. 

We do have functions `get_ulong_attribute_by_type()` and friends already in attributes.c, but including that in the usr/sbin/pkcsicsf/pkcsicsf tool causes similar problems.

Reply:
ok